lib/ansible/config/manager.py

```diff
@@ -521,6 +521,16 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                     raise AnsibleOptionsError('Invalid type for configuration option %s: %s' %
                                               (to_native(_get_entry(plugin_type, plugin_name, config)), to_native(e)))
 
+            # ensure choice is valid
+            if value is not None and defs[config].get('choices'):
+                if value not in defs[config].get('choices'):
+                    choices = defs[config].get('choices')
+                    if isinstance(choices, Mapping):
+                        choices = ', '.join(choices.keys())
```

Comment:
Shouldn't `choices = ', '.join(choices)` also work? I.e. is the `if` really necessary?

Reply:
A habit due to not all version of Python do the same thing for `iter` so the safe way was using explicit `keys()` .. which also does diff things but has less traps as ViewKeys and list copy of keys are not as incompatible.